File: CarSystemUI/app/src/main/java/com/android/car/systemui/navigation/BootReceiver.java
package com.android.car.systemui.navigation;

import android.content.BroadcastReceiver;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.app.WallpaperManager;
import android.os.Process;

import java.lang.reflect.Method;

import com.android.car.systemui.wallpaper.CarWallpaperService;

public final class BootReceiver extends BroadcastReceiver {
    private static final int PER_USER_RANGE = 100000;

    @Override
    public void onReceive(Context context, Intent intent) {
        if (Process.myUid() / PER_USER_RANGE != 0) {
            return;
        }

        context.startService(new Intent(context, BottomNavigationService.class));
        if (context.getSystemService(Context.WALLPAPER_SERVICE) == null) {
            return;
        }

        try {
            WallpaperManager wallpaperManager = WallpaperManager.getInstance(context);
            Method setWallpaperComponent = WallpaperManager.class.getMethod(
                    "setWallpaperComponent", ComponentName.class);
            setWallpaperComponent.invoke(
                    wallpaperManager, new ComponentName(context, CarWallpaperService.class));
        } catch (ReflectiveOperationException | SecurityException ignored) {
            // The platform image may already have selected the wallpaper for this user.
        }
    }
}

File: CarSystemUI/app/src/main/java/com/android/car/systemui/navigation/BootReceiver.kt
package com.android.car.systemui.navigation

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Process
import com.android.car.systemui.boot.BootHandoffService
import com.android.car.systemui.boot.BootUserPolicy
import com.android.car.systemui.di.SystemUiDependencies

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action !in SUPPORTED_ACTIONS) return
        if (!BootUserPolicy.isSystemUser(Process.myUid())) return
        if (BootUserPolicy.shouldShowHandoff(intent.action, Process.myUid())) {
            context.startService(BootHandoffService.showIntent(context))
        }
        context.startService(Intent(context, BottomNavigationService::class.java))
        SystemUiDependencies.from(context).startupRepository.initialize(context)
    }

    private companion object {
        val SUPPORTED_ACTIONS = setOf(
            Intent.ACTION_BOOT_COMPLETED,
            Intent.ACTION_LOCKED_BOOT_COMPLETED,
            Intent.ACTION_MY_PACKAGE_REPLACED,
        )
    }
}
